internal/app/parcel/repository_test.go

```diff
@@ -126,3 +126,90 @@ func TestRepository_FetchParcelByID(t *testing.T) {
 		assert.EqualError(t, err, "sql-error")
 	})
 }
+
+func TestRepository_UpdateParcel(t *testing.T) {
+	parcel := model.Parcel{
+		ID:                 1,
+		UserID:             1,
+		SourceAddress:      "Dhaka Bangladesh",
+		DestinationAddress: "Pabna Shadar",
+		SourceTime:         time.Now(),
+		ParcelType:         "Document",
+		Price:              200.0,
+		CarrierFee:         180.0,
+		CompanyFee:         20.0,
+		Status:             2,
+		CreatedAt:          time.Now(),
+		UpdatedAt:          time.Now(),
+	}
+
+	t.Run("should return success", func(t *testing.T) {
+		db, m, _ := sqlmock.New()
+		defer db.Close()
+
+		sqlxDB := sqlx.NewDb(db, "sqlmock")
+		m.ExpectExec("UPDATE parcel SET (.+) WHERE (.+)").
+			WithArgs(parcel.Status, parcel.ID).
+			WillReturnResult(sqlmock.NewResult(1, 1))
+
+		repo := NewRepository(sqlxDB)
+		err := repo.UpdateParcel(context.Background(), parcel)
+		assert.Nil(t, err)
+	})
+
+	t.Run("should return invalid ID", func(t *testing.T) {
+		db, m, _ := sqlmock.New()
+		defer db.Close()
+
+		sqlxDB := sqlx.NewDb(db, "sqlmock")
+		m.ExpectExec("UPDATE parcel SET (.+) WHERE (.+)").
+			WithArgs(parcel.Status, parcel.ID).
+			WillReturnResult(sqlmock.NewResult(1, 0))
```

Comment:
It will return error not result. the line should be like `WillReturnError(&pq.Error{Code: "23505"})`

Reply:
No, it will not return error. This is a case where you check if you req for an update with an ID which doesn't exist then it will return result with 0 rows. and for 0 rows check, it has custom invalid error.

Reply:
ooh, ok then that's alright.